lang/cs/Org.Apache.REEF.Bridge/JavaClrBridge.cpp

```diff
@@ -436,22 +436,35 @@ JNIEXPORT void JNICALL Java_org_apache_reef_javabridge_NativeInterop_clrSystemCo
   }
 }
 
+
 /*
- * Class:     org_apache_reef_javabridge_NativeInterop
- * Method:    clrSystemDriverRestartHandlerOnNext
- * Signature: (J)V
- */
-JNIEXPORT void JNICALL Java_org_apache_reef_javabridge_NativeInterop_clrSystemDriverRestartHandlerOnNext
-(JNIEnv *env , jclass cls, jlong handler) {
-  ManagedLog::LOGGER->Log("+Java_org_apache_reef_javabridge_NativeInterop_clrSystemDriverRestartHandlerOnNext");
-  try {
-    ClrSystemHandlerWrapper::Call_ClrSystemDriverRestart_OnNext(handler);
-  }
-  catch (System::Exception^ ex) {
-    String^ errorMessage = "Exception in Call_ClrSystemContextMessage_OnNext";
-    ManagedLog::LOGGER->LogError(errorMessage, ex);
-    // we do not call back to Java for exception in .NET restart handler
-  }
+* Class:     org_apache_reef_javabridge_NativeInterop
+* Method:    callClrSystemOnRestartHandlerOnNext
+* Signature: (Ljava/lang/String;Ljava/lang/String;Lorg/apache/reef/javabridge/EvaluatorRequestorBridge;)[J
+*/
+JNIEXPORT jlongArray JNICALL Java_org_apache_reef_javabridge_NativeInterop_callClrSystemOnRestartHandlerOnNext
+(JNIEnv * env, jclass jclassx, jstring dateTimeString, jstring httpServerPort, jobject jevaluatorRequestorBridge)
+{
+	try {
+		ManagedLog::LOGGER->Log("+Java_org_apache_reef_javabridge_NativeInterop_callClrSystemOnStartHandler");
+		const wchar_t* charConfig = UnicodeCppStringFromJavaString(env, dateTimeString);
+		int lenConfig = env->GetStringLength(dateTimeString);
+		String^  strConfig = Marshal::PtrToStringUni((IntPtr)(unsigned short*)charConfig, lenConfig);
+		DateTime dt = DateTime::Now;
+
+		const wchar_t* charPort = UnicodeCppStringFromJavaString(env, httpServerPort);
+		int lenPort = env->GetStringLength(httpServerPort);
+		String^  strPort = Marshal::PtrToStringUni((IntPtr)(unsigned short*)charPort, lenPort);
```

Comment:
Consider using ManagedStringFromJavaString. It will also call env->ReleaseStringChars